packages/neo-one-smart-contract-compiler/src/compile/syscalls.ts

```diff
@@ -1044,24 +1048,148 @@ export const SYSCALLS = {
   }),
   'Neo.Storage.Find': new SimpleSysCall({
     name: 'Neo.Storage.Find',
+    customDeclaration: `function syscall<T>(name: 'Neo.Storage.Find', value: Array<T>): Iterator<SerializableValue, SerializableValue>;`,
     args: [new SysCallArgument('context', StorageContextValue), new SysCallArgument('prefix', StorageKey)],
-    returnType: StorageIteratorValue,
-  }),
-  'Neo.Enumerator.Next': new SimpleSysCall({
-    name: 'Neo.Enumerator.Next',
-    args: [new SysCallArgument('iterator', StorageIteratorValue)],
-    returnType: BooleanValue,
-  }),
-  'Neo.Iterator.Key': new SimpleSysCall({
-    name: 'Neo.Iterator.Key',
-    args: [new SysCallArgument('iterator', StorageIteratorValue)],
-    returnType: StorageKey,
-  }),
-  'Neo.Enumerator.Value': new SimpleSysCall({
-    name: 'Neo.Enumerator.Value',
-    args: [new SysCallArgument('iterator', StorageIteratorValue)],
-    returnType: StorageValue,
+    returnType: IteratorValue,
   }),
+  'Neo.Enumerator.Create': (() => {
+    const name = 'Neo.Enumerator.Create';
+    const returnType = EnumeratorValue;
+
+    return {
+      name,
+      toDeclaration: () => `function syscall<T>(name: '${name}', value: Array<T>): EnumeratorBase<T>;`,
+      handleCall: (sb: ScriptBuilder, node: CallExpression, optionsIn: VisitOptions) => {
+        const options = sb.pushValueOptions(sb.noCastOptions(optionsIn));
+        // [arrayVal]
+        sb.visit(node.getArguments()[1], options);
+        // [array]
+        sb.emitHelper(node, options, sb.helpers.unwrapArray);
+        // [Neo.Enumerator.Create, array]
+        sb.emitSysCall(node, name);
+
+        if (optionsIn.pushValue) {
+          returnType.handleResult(sb, node, options, optionsIn.cast);
+        } else {
+          sb.emitOp(node, 'DROP');
+        }
+      },
+    };
+  })(),
+
+  'Neo.Enumerator.Next': (() => {
+    const name = 'Neo.Enumerator.Next';
+    const returnType = BooleanValue;
+    const valueType = IteratorValue;
+
+    return {
```

Comment:
This should be using `SimpleSysCall`, you don't need a custom `handleCall` implementation